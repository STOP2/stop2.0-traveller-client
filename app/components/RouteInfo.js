import React, { Component } from 'react'
import { View, Image } from 'react-native'

import {DefaultText} from '../components/textComponents'

import styles from '../styles/stylesheet'
import strings from '../resources/translations'

class RouteInfo extends Component {
    constructor(props)
    {
        super(props)

        this.state = {}
        this.iconTram = require('../resources/icons/hsl_reittiopas_tram.png')
        this.iconBus = require('../resources/icons/hsl_reittiopas_bus.png')
    }

    render()
    {
        return (<View style={styles.stopRequestStyle}>
          <View style={styles.stopRequestVehicleInfo}>
            <Image style={styles.busIcon} resizeMode="contain"
            source={this.props.vehicleType == 0 ? this.iconTram : this.iconBus}/>
            <View style={styles.vehicleInfoWrapper}>
              <DefaultText style={styles.vehicleInfo}>{this.props.vehicleLine} {this.props.vehicleDestination}</DefaultText>
            </View>
          </View>
          <View style={styles.vehicleArrivesInWrapper}>
            <DefaultText style={styles.vehicleArrivesInText}>{strings.vehicleArrivesIn}</DefaultText>
          </View>
          <View>
            <DefaultText style={styles.vehicleMinutesLeft}>{this.props.vehicleMinutesLeft}</DefaultText>
          </View>
        </View>)
    }
}

RouteInfo.propTypes = {
    title: React.PropTypes.string.isRequired,
    vehicleType: React.PropTypes.number.isRequired,
    vehicleLine: React.PropTypes.string.isRequired,
    vehicleDestination: React.PropTypes.string.isRequired,
    vehicleMinutesLeft: React.PropTypes.string.isRequired
}

export default RouteInfo
