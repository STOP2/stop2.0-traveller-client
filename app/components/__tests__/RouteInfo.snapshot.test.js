import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import RouteInfo from '../RouteInfo';

jest.mock('react-native-localization');

describe('RouteInfo component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RouteInfo title="title" vehicleType={0} vehicleLine="506" vehicleDestination="Meilahti" vehicleMinutesLeft="5 min" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
