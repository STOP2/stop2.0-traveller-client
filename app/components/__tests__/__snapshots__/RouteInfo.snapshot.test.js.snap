exports[`RouteInfo component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "gotham-rounded-book",
          },
          Object {
            "fontSize": 40,
            "textAlign": "center",
          },
        ]
      }>
      title
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 20,
      }
    }>
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      } />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "gotham-rounded-book",
            },
            Object {
              "flex": 1,
              "fontSize": 20,
              "textAlign": "center",
            },
          ]
        }>
        506
         
        Meilahti
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 2,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "gotham-rounded-book",
          },
          Object {
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }>
      Vehicle arrives in: 
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "gotham-rounded-book",
          },
          Object {
            "fontSize": 40,
            "textAlign": "center",
          },
        ]
      }>
      5 min
    </Text>
  </View>
</View>
`;
